overlord/devicestate/devicestate_install_mode_test.go

```diff
@@ -369,6 +369,104 @@ func (s *deviceMgrInstallModeSuite) TestInstallTaskErrors(c *C) {
 	c.Check(s.restartRequests, HasLen, 0)
 }
 
+func (s *deviceMgrInstallModeSuite) TestInstallExpTasks(c *C) {
+	restore := release.MockOnClassic(false)
+	defer restore()
+
+	restore = devicestate.MockInstallRun(func(mod gadget.Model, gadgetRoot, kernelRoot, device string, options install.Options, _ gadget.ContentObserver) (*install.InstalledSystemSideData, error) {
+		return nil, nil
+	})
+	defer restore()
+
+	err := ioutil.WriteFile(filepath.Join(dirs.GlobalRootDir, "/var/lib/snapd/modeenv"),
+		[]byte("mode=install\n"), 0644)
+	c.Assert(err, IsNil)
+
+	s.state.Lock()
+	s.makeMockInstalledPcGadget(c, "dangerous", "")
+	devicestate.SetSystemMode(s.mgr, "install")
+	s.state.Unlock()
+
+	s.settle(c)
+
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	installSystem := s.findInstallSystem()
+	c.Check(installSystem.Err(), IsNil)
+
+	tasks := installSystem.Tasks()
+	c.Assert(tasks, HasLen, 2)
+	setupRunSystemTask := tasks[0]
+	ensureNextBootToRunModeTask := tasks[1]
+
+	c.Assert(setupRunSystemTask.Kind(), Equals, "setup-run-system")
+	c.Assert(ensureNextBootToRunModeTask.Kind(), Equals, "ensure-next-boot-to-run-mode")
+
+	// setup-run-system has no pre-reqs
+	c.Assert(setupRunSystemTask.WaitTasks(), HasLen, 0)
+
+	// ensure-next-boot-to-run-mode has a pre-req of setup-run-system
+	waitTasks := ensureNextBootToRunModeTask.WaitTasks()
+	waitTaskIDs := make([]string, 0, len(waitTasks))
```

Comment:
this feels a bit complicated given we expect one task, wouldn't it be simpler to check HasLen, 1  and then compare waitTasks[0].ID() ?